status/status.go

```diff
@@ -16,70 +16,188 @@ limitations under the License.
-type Reporter interface {
-	ReportStatusForSnapshot(client.Client, context.Context, *helpers.IntegrationLogger, *applicationapiv1alpha1.Snapshot) error
+// generate TestReport to be used by all reporters
+func generateTestReport(ctx context.Context, client client.Client, detail intgteststat.IntegrationTestStatusDetail, snapshot *applicationapiv1alpha1.Snapshot) (*TestReport, error) {
+	text, err := generateText(client, ctx, detail, snapshot.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate text message: %w", err)
+	}
+
+	summary, err := generateSummary(detail.Status, snapshot.Name, detail.ScenarioName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate summary message: %w", err)
+	}
+
+	report := TestReport{
+		Text:           text,
+		FullName:       NamePrefix + " / " + snapshot.Name + " / " + detail.ScenarioName,
+		ScenarioName:   detail.ScenarioName,
+		Status:         detail.Status,
+		Summary:        summary,
+		StartTime:      detail.StartTime,
+		CompletionTime: detail.CompletionTime,
+	}
+	return &report, nil
 }
 
-// Status is the interface of the main status Adapter.
-type Status interface {
-	GetReporters(client.Object) ([]Reporter, error)
+type StatusInterface interface {
+	GetReporter(*applicationapiv1alpha1.Snapshot) ReporterInterface
+	ReportSnapshotStatus(context.Context, ReporterInterface, *applicationapiv1alpha1.Snapshot) error
 }
 
-// Adapter is responsible for discovering supported Reporter implementations.
-type Adapter struct {
-	logger         logr.Logger
-	k8sClient      client.Reader
-	githubReporter Reporter
+type Status struct {
+	logger logr.Logger
+	client client.Client
 }
 
-// AdapterOption is used to extend Adapter with optional parameters.
-type AdapterOption = func(a *Adapter)
+// check if interface has been implemented correctly
+var _ StatusInterface = (*Status)(nil)
 
-// WithGitHubReporter is an option which allows for replacement of the GitHub PipelineRun reporter.
-func WithGitHubReporter(reporter Reporter) AdapterOption {
-	return func(a *Adapter) {
-		a.githubReporter = reporter
+func NewStatus(logger logr.Logger, client client.Client) *Status {
+	return &Status{
+		logger: logger,
+		client: client,
 	}
 }
 
-// NewAdapter constructs an Adapter with optional params, if specified.
-func NewAdapter(logger logr.Logger, k8sClient client.Client, opts ...AdapterOption) *Adapter {
-	adapter := Adapter{
-		logger:         logger,
-		k8sClient:      k8sClient,
-		githubReporter: NewGitHubReporter(logger, k8sClient),
+// Get reporter which is able to process snapshot, nil means no suitable reporter found
+func (s *Status) GetReporter(snapshot *applicationapiv1alpha1.Snapshot) ReporterInterface {
+	githubReporter := NewGitHubReporter(s.logger, s.client)
+	if githubReporter.Detect(snapshot) {
+		return githubReporter
 	}
 
-	for _, opt := range opts {
-		opt(&adapter)
+	return nil
+}
+
+// Report status of all integration tests into Pull Request
```

Comment:
Tiny nit: should start with `ReportSnapshotStatus`

Reply:
updated